src/trustyai/metrics/saliency.py

```diff
@@ -0,0 +1,136 @@
+
+
+def mean_impact_score(explainer, model, data, is_model_callable=False, k=2):
+    """
+    Parameters
+    ----------
+    explainer: Union[trustyai.explainers.LimeExplainer, trustyai.explainers.SHAPExplainer]
+        the explainer to evaluate
+    model: trustyai.PredictionProvider
+        the model used to generate predictions
+    data: list[list[trustyai.model.Feature]]
+        the inputs to calculate the metric for
+    is_model_callable: bool
+        whether to directly use model function call or use the predict method
+    k: int
+        the number of top important features
+
+    Returns
+    -------
+    :float:
+        the mean impact score metric across all inputs
+    """
+    m_is = 0
+    for features in data:
+        m_is += impact_score(model, features, explainer, k, is_model_callable=is_model_callable)
+    return m_is / len(data)
+
+
+def classification_fidelity(explainer, model, inputs, is_model_callable=False):
+    """
+    Parameters
+    ----------
+    explainer: Union[trustyai.explainers.LimeExplainer, trustyai.explainers.SHAPExplainer]
+        the explainer to evaluate
+    model: trustyai.PredictionProvider
+        the model used to generate predictions
+    inputs: list[list[trustyai.model.Feature]]
+        the inputs to calculate the metric for
+    is_model_callable: bool
+        whether to directly use model function call or use the predict method
+
+    Returns
+    -------
+    :float:
+        the classification fidelity metric
+    """
+    pairs = []
+    for c_input in inputs:
+        if is_model_callable:
+            output = model(c_input)
+        else:
+            output = model.predict([c_input])[0].outputs
+        explanation = explainer.explain(inputs=c_input, outputs=output, model=model)
+        saliency = list(explanation.saliency_map().values())[0]
+        pairs.append(_Pair.of(saliency, simple_prediction(c_input, output)))
+    return ExplainabilityMetrics.classificationFidelity(pairs)
+
+
+def local_saliency_f1(output_name, model, explainer, distribution, k, chunk_size):
+    """
+    Parameters
+    ----------
+    output_name: str
+        the name of the output to calculate the metric for
+    model: trustyai.PredictionProvider
+        the model used to generate predictions
+    explainer: Union[trustyai.explainers.LIMEExplainer, trustyai.explainers.SHAPExplainer,
+                trustyai.explainers.LocalExplainer]
+        the explainer to evaluate
+    distribution: org.kie.trustyai.explainability.model.PredictionInputsDataDistribution
+        the data distribution to fetch the inputs from
+    k: int
+        the number of top important features
+    chunk_size: int
+        the chunk of inputs to fetch fro the distribution
+
+    Returns
+    -------
+    :float:
+        the local saliency f1 metric
+    """
+    if isinstance(explainer, LimeExplainer):
+        local_explainer = LocalExplainer(explainer._explainer)
```

Comment:
I haven't tested it, but I wonder if casting to the interface directly with JPype would remove the need to re-implement the `LocalExplainer` "type".

For instance:

```python
from org.kie.trustyai.explainability.local import LocalExplainer

local_explainer = JObject(explainer._explainer, LocalExplainer)
```

Reply:
well, thanks  @ruivieira that's much better and avoids introducing extra class needlessly.